api/run.go

```diff
@@ -0,0 +1,81 @@
+// Package api allows go projects to use func-e as a library.
+package api
+
+import (
+	"github.com/tetratelabs/func-e/internal/cmd"
+	"github.com/tetratelabs/func-e/internal/globals"
+	"github.com/tetratelabs/func-e/internal/version"
+	"io"
+	"os"
+	"runtime"
+)
+
+// HomeDir is an absolute path which most importantly contains "versions"
+// installed from EnvoyVersionsURL. Defaults to "${HOME}/.func-e"
+func HomeDir(homeDir string) RunOption {
+	return func(o *runOpts) {
+		o.homeDir = homeDir
+	}
+}
+
+// EnvoyVersionsURL is the path to the envoy-versions.json.
+// Defaults to "https://archive.tetratelabs.io/envoy/envoy-versions.json"
+func EnvoyVersionsURL(envoyVersionsURL string) RunOption {
```

Comment:
ex unit tests can set this to httptest server

Reply:
not sure what you mean here

Reply:
cool makes sense with https://github.com/tetratelabs/func-e/pull/434#issuecomment-1295587586